src/hdusd/engine/viewport_engine.py

```diff
@@ -294,18 +297,21 @@ def draw(self, context):
         bgl.glBlendFunc(bgl.GL_ONE, bgl.GL_ONE_MINUS_SRC_ALPHA)
         self.render_engine.bind_display_space_shader(context.scene)
 
+        if self.renderer.GetRenderStats()['percentDone'] == 0.0:
+            self.time_begin = time.perf_counter()
+
         try:
             self.renderer.Render(stage.GetPseudoRoot(), self.render_params)
         except Exception as e:
             log.error(e)
 
         self.render_engine.unbind_display_space_shader()
         bgl.glDisable(bgl.GL_BLEND)
-
+        elapsed_time = time_str(time.perf_counter() - self.time_begin)
         if not self.renderer.IsConverged():
-            self.notify_status("Rendering...", "")
+            self.notify_status(f"Render Time: {elapsed_time} | Done: {round(self.renderer.GetRenderStats()['percentDone'])}%", "Rendering...")
```

Comment:
Better to use following notification:
self.notify_status(f"Time: {elapsed_time} | Done: {round(self.renderer.GetRenderStats()['percentDone'])}%", "Render")